instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/ClassNames.java

```diff
@@ -7,21 +7,29 @@
 
 public class ClassNames {
 
+  private static final ClassValue<String> simpleNames =
+      new ClassValue<String>() {
+        @Override
+        protected String computeValue(Class<?> type) {
+          if (!type.isAnonymousClass()) {
+            return type.getSimpleName();
+          }
+          String className = type.getName();
+          if (type.getPackage() != null) {
```

Comment:
Packages are optional. Perhaps it would be better to use the same strategy as jdk uses
```
className = className.substring(className.lastIndexOf('.') + 1);
```